modules/apps/transfer/types/token_test.go

```diff
@@ -149,3 +151,50 @@ func TestValidate(t *testing.T) {
 		})
 	}
 }
+
+func TestToCoin(t *testing.T) {
+	testCases := []struct {
+		name     string
+		token    Token
+		coins    sdk.Coin
+		expError error
+	}{
+		{
+			"success: convert token to coin",
+			Token{
+				Denom: Denom{
+					Base:  denom,
+					Trace: []Trace{},
+				},
+				Amount: amount,
+			},
+			sdk.NewCoin(denom, sdkmath.NewInt(100)),
+			nil,
+		},
+		{
+			"failure: invalid amount string",
+			Token{
+				Denom: Denom{
+					Base:  denom,
+					Trace: []Trace{},
+				},
+				Amount: "value",
+			},
+			sdk.Coin{},
+			ErrInvalidAmount,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			coin, err := tc.token.ToCoin()
+			expPass := tc.expError == nil
+			if expPass {
+				require.NoError(t, err, tc.name)
+				require.Equal(t, tc.coins, coin, tc.name)
+			} else {
+				require.ErrorContains(t, err, tc.expError.Error(), tc.name)
```

Comment:
You could also do here:

```go
require.Equal(t, tc.coins, coin, tc.name)
```

to check that the coin returned in the failure case is empty.

Reply:
Thank you!